regiontag_test.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2016 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+package samples
+
+import (
+	"bufio"
+	"bytes"
+	"log"
+	"os/exec"
+	"strings"
+	"testing"
+)
+
+func TestRegionTags(t *testing.T) {
+	type RegionTag struct {
+		tag  string
+		path string
+	}
+	errs := []RegionTag{}
+
+	cmd := exec.Command("go", "list", "./...")
+	out, err := cmd.Output()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	scanner := bufio.NewScanner(strings.NewReader(string(out)))
+	for scanner.Scan() {
+		path := scanner.Text()
+
+		goDoc := exec.Command("go", "doc", path)
+		grep := exec.Command("grep", "START")
+		grep.Stdin, _ = goDoc.StdoutPipe()
+
+		var buf bytes.Buffer
+		writer := bufio.NewWriter(&buf)
+		grep.Stdout = writer
+
+		grep.Start()
+		goDoc.Run()
+		grep.Wait()
+
+		writer.Flush()
+		s := buf.String()
+		if len(s) > 0 {
+			errs = append(errs, RegionTag{s, path})
+		}
+	}
+
+	if len(errs) > 0 {
```

Comment:
I wonder if the errors could be printed as they're found, rather than accumulating in the `errs` slice? Might be able to get rid of the `RegionTag` type, too.


```go
t.Errorf("Region tag %q found in GoDoc of %q", s, path)
```

Reply:
That was a leftover of a previous async implementation that failed. I've made a few changes and got the async version working so I think now it justify doing the things the way it's done... but if you think we should remove the type and handle things differently please let me know and I'll fix it.